packages/loot-core/src/server/accounts/link.js

```diff
@@ -127,6 +127,7 @@ export async function addNordigenAccounts(
       let id = await runMutator(async () => {
         let id = await db.insertAccount({
           account_id: acct.account_id,
```

Comment:
1. Could the IBAN be used as `account_id` instead?
2. How sensitive is the IBAN? Do you think it‘s ok to permanently record it in the budget like this?

Reply:
> 1. Could the IBAN be used as `account_id` instead?

Short answer: `account_id` can be used to fetch a single account through the API endpoints. IBAN can't (unless it happens to be equal to `account_id`).

Long answer: I wouldn't advise it. I don't know how banks and Nordigen would handle such a rare scenario but theoretically, it's not guaranteed that the same account couldn't change an IBAN - I know of an institution here in Poland that had to migrate their IBANs due to some legal changes. During the migration period, both IBANs were valid but who knows what a bank integration would have returned if those were a thing back then. Maybe it would just require creating a new agreement with Nordigen and this wouldn't be a problem; but *maybe*, this number would change causing something to break.

> 2\. How sensitive is the IBAN? Do you think it‘s ok to permanently record it in the budget like this?

It's not *very* sensitive since one should be able to freely share it with anyone that wants to make a transfer to them and as such only allows you to receive transfers, not make them. However, it is not a completely random identifier and what it's based on depends on the country. There's some info here:
https://money.stackexchange.com/questions/90674/is-it-really-risky-to-make-your-iban-public

I personally think that it's fine to store it, especially when Actual docs recommend enabling encryption when using bank sync. The alternative could be fetching IBANs for **all** accounts before every single `/transactions` call. I suppose that since the basic account metadata API is not affected by bank rate limits, it wouldn't necessarily be unreasonable, especially with some short-lived cache on the server's side.